pluginLoader.js

```diff
@@ -33,7 +33,12 @@ define(function(require, exports, module){
             var _lint = plugin.lint;
 
             function lint(text, settings) {
-                return Promise.resolve(_lint(text, settings));
+                var results = _lint(text, settings,pluginsMeta);
+                if(results.promise !== undefined && results.promise !== null){
```

Comment:
So I'm checking to see if the `lint()` method returns a promise or a standard object.

At line 15 in plugins/default/embedded/php/main.js I setup a Promise to be returned - the Promise object has a `promise` property /shrug it should probably be checking in a better way I guess.

I needed to use a promise as the `child_process.exec()` method is async - so the results of the process would not be available when the `lint()` method @ line 7 returns. So then I followed the data flow (which crossed my eyes a few times) and at this point ^^ you are resolving the Promise with the lint results from whichever plugin is returning. So instead I passed the Promise through.

does that explain it?